pkg/controller/acmeorders/util.go

```diff
@@ -122,7 +122,8 @@ func partialChallengeSpecForAuthorization(ctx context.Context, issuer cmapi.Gene
 	}
 
 	// 2. filter solvers to only those that matchLabels
-	for _, cfg := range solvers {
+	for i := range solvers {
+		cfg := solvers[i]
```

Comment:
Not sure about this one. DeepCopy seems to be used below, but the body of this for loop is far too long and complex to know for sure.